apps/desktop-wallet/src/api/queries/addressQueries.ts

```diff
@@ -132,3 +135,33 @@ export const addressTokensSearchStringQuery = ({ addressHash, networkId }: Addre
       }
     }
   })
+
+export const addressTokensByTypeQuery = ({ addressHash, networkId }: AddressLatestTransactionQueryProps) =>
+  queryOptions({
+    queryKey: ['address', addressHash, 'computedData', 'tokensByType', { networkId }],
+    ...getQueryConfig({ staleTime: Infinity, gcTime: Infinity, networkId }),
+    queryFn: async () => {
+      const { balances: addressAlphBalances } = await queryClient.fetchQuery(
+        addressAlphBalancesQuery({ addressHash, networkId })
+      )
+      const { balances: addressTokensBalances } = await queryClient.fetchQuery(
+        addressTokensBalancesQuery({ addressHash, networkId })
+      )
+      const allTokensBalances =
+        addressAlphBalances && addressAlphBalances.totalBalance !== '0'
+          ? [{ id: ALPH.id, ...addressAlphBalances } as TokenApiBalances, ...addressTokensBalances]
+          : addressTokensBalances
+
+      const ftList = await queryClient.fetchQuery(ftListQuery({ networkId }))
+      const { listedFts, unlistedTokens } = separateTokensByListing(allTokensBalances, ftList)
+
+      const tokenTypesPromiseResults = await Promise.allSettled(
+        unlistedTokens.map(({ id }) => queryClient.fetchQuery(tokenTypeQuery({ id, networkId })))
+      )
+      const tokenTypes = getFulfilledValues(tokenTypesPromiseResults)
+
+      const { fungible: unlistedFtIds, 'non-fungible': nftIds, 'non-standard': nstIds } = combineTokenTypes(tokenTypes)
+
+      return { listedFts, unlistedTokens, unlistedFtIds, nftIds, nstIds }
+    }
+  })
```

Comment:
Instead of aggregating all this data in a hook (which can be expensive when there are many addresses and cannot be cached), aggregating them in a `queryFn` in combination with `fetchQuery` (important!) allows us to benefit from the Tanstack caching mechanism.